src/apprt/gtk/ActionPalette.zig

```diff
@@ -0,0 +1,692 @@
+
+    var it = std.mem.tokenizeScalar(u8, self.query, ' ');
+    self.tokens.clearRetainingCapacity();
+
+    while (it.next()) |token| {
+        self.tokens.append(self.alloc, token) catch @panic("OOM");
+    }
+
+    log.warn("search changed={s}", .{self.tokens.items});
+
+    self.filter.changed(.different);
+    self.sorter.changed(.different);
+    // self.selection.setSelected(0);
+
+    const scroll_info = gtk.ScrollInfo.new();
+    scroll_info.setEnableVertical(1);
+    self.list_view.scrollTo(0, .{ .select = true }, scroll_info);
+}
+fn searchActivated(_: *gtk.SearchEntry, self: *ActionPalette) callconv(.C) void {
+    // Activate the first action
+    self.activate(self.selection.getSelected());
+}
+
+fn itemFactorySetup(_: *gtk.SignalListItemFactory, object: *gobject.Object, _: *ActionPalette) callconv(.C) void {
+    const item = gobject.ext.cast(gtk.ListItem, object).?;
+    const row = gtk.CenterBox.new();
+
+    const title = gtk.Label.new("").as(gtk.Widget);
+    title.setMarginTop(10);
+    title.setMarginBottom(10);
+    title.setMarginStart(10);
+    title.setMarginEnd(10);
+
+    const bind = gtk.ShortcutLabel.new("").as(gtk.Widget);
+    bind.setMarginTop(10);
+    bind.setMarginBottom(10);
+    bind.setMarginStart(10);
+    bind.setMarginEnd(10);
+
+    row.setStartWidget(title);
+    row.setEndWidget(bind);
+
+    item.setChild(row.as(gtk.Widget));
+}
+fn itemFactoryBind(_: *gtk.SignalListItemFactory, object: *gobject.Object, self: *ActionPalette) callconv(.C) void {
+    const item = gobject.ext.cast(gtk.ListItem, object).?;
+    const row = gobject.ext.cast(gtk.CenterBox, item.getChild().?).?;
+    const cmd = Command.fromEnumListItem(item.getItem());
+
+    const title = gobject.ext.cast(gtk.Label, row.getStartWidget().?).?;
+    title.setText(cmd.title());
+
+    const bind = gobject.ext.cast(gtk.ShortcutLabel, row.getEndWidget().?).?;
+    const trigger = self.window.app.config.keybind.set.getTrigger(cmd.action());
+
+    if (trigger) |t| {
+        var buf: [256]u8 = undefined;
+        const accel = key.accelFromTrigger(&buf, t) catch return;
+        if (accel) |a| bind.setAccelerator(a);
+    }
+}
+fn itemFactoryUnbind(_: *gtk.SignalListItemFactory, object: *gobject.Object, _: *ActionPalette) callconv(.C) void {
+    const item = gobject.ext.cast(gtk.ListItem, object).?;
+    const row = gobject.ext.cast(gtk.CenterBox, item.getChild().?).?;
+
+    const title = gobject.ext.cast(gtk.Label, row.getStartWidget().?).?;
+    const bind = gobject.ext.cast(gtk.ShortcutLabel, row.getEndWidget().?).?;
+
+    title.setText("");
+    bind.setAccelerator("");
+}
+fn itemFactoryTeardown(_: *gtk.SignalListItemFactory, object: *gobject.Object, _: *ActionPalette) callconv(.C) void {
+    const item = gobject.ext.cast(gtk.ListItem, object).?;
+    item.setChild(null);
+}
+
+/// A selection of commands/actions that actually make sense in a GUI context
+const Command = enum(c_int) {
```

Comment:
Looking at the shape of this, we can and should definitely generate this via comptime in core. I can take a stab at it. I don't see any reason this would be a burden and I think my approach as suggested would be _close_ to what works here.

libghostty also requires that any enum that makes it out into C ABI space is backed by C types and all strings are null-terminated so even if some mangling is needed in the GTK apprt, its probably minimal and now that I see how this works I can help shape it to that anyways.